### ISSUE TITLE
raise ValueError(f"No gradients provided for any variable: {variable}.

### ISSUE THREAD

**Issue by hammadamjad** [tensorflow/tensorflow#56909]
```
all_vars = [var for var in tf.compat.v1.trainable_variables()
            if 'g_' in var.name]
all_opt = tf.optimizers.Adam(learning_rate=0.0001).minimize(
    lossDict["all_loss"], var_list=all_vars, tape=tf.GradientTape())
for var in tf.trainable_variables():
    print("Listing trainable variables ... ", var)

saver = tf.train.Saver(max_to_keep=20)
config = tf.ConfigProto()
config.gpu_options.allow_growth = True
sess = tf.Session(config=config)
sess.run(tf.global_variables_initializer())
var_restore = [v for v in tf.trainable_variables()]
saver_restore = tf.train.Saver(var_restore)
ckpt = tf.train.get_checkpoint_state('result/'+task)
```

this is my code and i am facing no gradient issue...

**Comment by sushreebarsa** [comment]
@hammadamjad 
In order to expedite the trouble-shooting process here,Could you please fill the issue [template](https://github.com/tensorflow/tensorflow/issues/new/choose),
Thank you!
